tests-legacy/Unit/Controller/FrontController/ProductControllerTest.php

```diff
@@ -39,20 +46,44 @@ class ProductControllerTest extends IntegrationTestCase
 
     private $controller;
 
+    public static function setUpBeforeClass()
+    {
+        parent::setUpBeforeClass();
+        self::installTestedLanguagePacks();
+    }
+
     protected function setUp()
     {
         parent::setUp();
-        $this->contextMocker = new ContextMocker();
-        $this->contextMocker->mockContext();
-        $this->controller = new \ProductControllerCore();
+        global $kernel;
```

Comment:
oups I think I vomited a bit 🤢
but I guess you don't have much choice?

Reply:
vomited much more than you when writing this :/